test/helpers/cmd/reader.go

```diff
@@ -142,3 +146,28 @@ func TailReader(pod *corev1.Pod, path string) (*Reader, error) {
 func FileReader(pod *corev1.Pod, path string) (*Reader, error) {
 	return NewReader(runtime.Exec(pod, "cat", path))
 }
+
+// readErr if err != nil close and return a combined read+exit error.
+func (r *Reader) readErr(err error) error {
+	if err != nil {
+		if err2 := r.Close(); err2 != nil {
+			return fmt.Errorf("%v: %w", err, err2)
+		}
+	}
+	return err
+}
+
+const stderrLimit = 1024
+
+type stderrBuffer struct{ b bytes.Buffer }
+
+func (b *stderrBuffer) Write(data []byte) (int, error) {
+	max := stderrLimit - b.b.Len()
+	if max < 0 {
+		max = 0
```

Comment:
Would it make sense to just return here instead of going with a degenerate case of max == 0?